src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java

```diff
@@ -26,38 +26,34 @@
 import ca.corefacility.bioinformatics.irida.model.sample.StaticMetadataTemplateField;
 import ca.corefacility.bioinformatics.irida.model.sample.metadata.MetadataEntry;
 import ca.corefacility.bioinformatics.irida.ria.web.components.agGrid.AgGridColumn;
-import ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.UIMetadataField;
-import ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.UIMetadataFieldDefault;
-import ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.UIMetadataTemplate;
-import ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.UISampleMetadata;
-import ca.corefacility.bioinformatics.irida.security.permissions.project.ProjectOwnerPermission;
-import ca.corefacility.bioinformatics.irida.security.permissions.sample.UpdateSamplePermission;
+import ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.*;
 import ca.corefacility.bioinformatics.irida.service.ProjectService;
 import ca.corefacility.bioinformatics.irida.service.sample.MetadataTemplateService;
 import ca.corefacility.bioinformatics.irida.service.sample.SampleService;
 
+import com.google.common.collect.ImmutableMap;
+
 /**
  * This controller is responsible for AJAX handling for the line list page, which displays sample metadata.
  */
 @Controller
+@Scope("session")
 @RequestMapping("/linelist")
 public class LineListController {
 	private ProjectService projectService;
 	private SampleService sampleService;
+	private LineListPermissions permissions;
```

Comment:
Does this need to be a property on the class?  Every time it gets used you're calling `this.permissions.setPrincipalPermissions(principal);` first so it doesn't seem to need to be a property.

Reply:
I will look at reworking this.

Reply:
This was set up to be autowired because the component itself is autowired with services.  I suppose I could autowire those dependencies straight into the controller and have private methods inside the controller, I was just trying to clean the code up a bit and make this its own component.

Reply:
Looking into it a bit more I'm understanding what you're doing with it.  Having the permissions functions in their own component is fine, but something feels off about the way it's being used.  If it was just a single method call like `permissions.canModifySample(principal, psj.getObject())` then it would be no issue.  Calling the `this.permissions.setPrincipalPermissions(principal);` is not a good idea because that's a shared object between all users logged into the system. If you could use those functions without the call first to `setPrincipalPermissions` I think it would be better.

Reply:
This should be fixed now